### PR TITLE
add query runner support for query file resampling

### PR DESCRIPTION
**Pull request by apucher**
## Description
We add support for resampling of the query input file to the pinot-tools QueryRunner. This enables us to easily expand a small set of queries proportionally to run performance benchmarks at scale. Also adds tracking of query exceptions. All changes are fully backwards-compatible.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
**No**

Does this PR fix a zero-downtime upgrade introduced earlier?
**No**

Does this PR otherwise need attention when creating release notes? Things to consider:
**No**

### CODE REVIEW

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java [hunk 1]
```diff
@@ -49,6 +53,10 @@
   private String _mode;
   @Option(name = "-queryFile", required = true, metaVar = "<String>", usage = "Path to query file.")
   private String _queryFile;
+  @Option(name = "-queryMode", required = false, metaVar = "<String>", usage = "Mode of query generator (list|sample).")
```
Comment (fx19880617) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:56]: full|sample?

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java [hunk 2]
```diff
@@ -622,6 +632,22 @@ public static void increasingQPSQueryRunner(PerfBenchmarkDriverConf conf, String
...
 
+  private static Stream<String> makeQueries(List<String> inputs, QueryMode queryMode, int queryCount) {
+    queryCount = queryCount > 0 ? queryCount : inputs.size();
+
+    switch (queryMode) {
+      case LIST:
+        return inputs.stream().limit(queryCount);
+
+      case SAMPLE:
+        Random r = new Random(inputs.hashCode()); // anything deterministic will do
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:643]: This hashCode could be quite expensive, maybe just use `new Random(0)`?

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java (hunk 1 above)
Comment (apucher) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:56]: `full|resample` maybe

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java (hunk 2 above)
Comment (apucher) [reply pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:643]: sure. since this is a one-off shouldn't really impact perf at all

**Pull request merged**